src/index.js

```diff
@@ -45,3 +45,22 @@ exports.Queue = Plotly.Queue;
 
 // export d3 used in the bundle
 exports.d3 = require('d3');
+
+Plotly.register({
+    traces: {
+        bar: require('./traces/bar'),
```

Comment:
@bpostlethwaite 

> it would be nice to make it clear what is registerable though.
> What makes a component registerable?

Maybe we store another meta property in the trace modules? e.g.

```js
var PlotlyBar = require('plotly.js/lib/bar');

console.log(PlotlyBar.moduleType);  // => 'trace'
```

and down the road, registerable component module will have `moduleType` set to `'component'`.

Moreover, `Plotly.register` should error out if given a module without `moduleType`.